### ISSUE TITLE
Slow download speeds

### ISSUE THREAD

**Issue by Ackr1** [eviabs/downtify-premium#31]
I apologize if this should be obvious, but is the download speed supposed to be extremely slow? (about 2/3 of the actual playtime of a track in my case, and I am on gigabit fiber)

Is the program "recording" and re-encoding the tracks, and if so, does that not result in a loss in quality? 

If there is nothing to do about download speeds, would it be possible to do simultanous downloads of multiple tracks or is that impossible without multiple premium accounts?

**Comment by leetfin** [comment]
Yes, as far as I know it is recording and encoding the track, I don't think anything can be done about it but I'd love to be wrong.

**Comment by eviabs** [comment]
Yeah, the program records the songs and re-encoding them.
Unfortunately, I'm not aware of a better solution (Spotify music files are encrypted or use unknown format).

The program uses your local Spotify installation API to play and control the music. Since only one device can play music, we can't run this simultaneously.

I didn't try to run multiple instances with different accounts, maybe it would work (you can probably use sandboxie for that).

**Comment by Ackr1** [comment]
I understand, thank you for taking the time to explain.

Compared to other solutions that record from the soundcard or download tracks from youtube and metadata from spotify, do you reckon using the API in the way your program does result in higher quality?

**Comment by spfy13** [comment]
I doubt YouTube would have better quality than Spotify. The highest quality streaming service that could be recorded would be Apple Music Lossless but no one has made anything that does that and I'm not sure it would be possible they may have a strong lock on their music and encryption.

**Comment by Ackr1** [comment]
I was referring more to the method than the source. YouTube downloaders do not "record" as the stream is not encrypted, they just strip the audio and convert/encode to another format. So even if youtube as a source is lower quality, I wonder how the end result compares to downify and other methods (that also "records").
